File: HoWo7/Phone/Smartphone.py
from Phone.Phone import Phone


class Smartphone(Phone):
    typeOfAntenna: str
    OS: str

    def __init__(self, model: str, sizeDisplay, akb: int, price, typeOfAntenna: str, os: str):
        super().__init__(model, sizeDisplay, akb, price)
        self.typeOfAntenna = typeOfAntenna
        self.OS = os

    def loginToTheInternet():
        print("вы вошли в интернет, прощайте)")

    def downloadApp():
        print("очередная шляпа установлена")

    def getModel(self) -> str:
        return super().getModel()

    def getSizeDisplay(self):  # -> double:
        return super().getSizeDisplay()

    def getAkb(self) -> int:
        return super().getAkb()

    def getPrice(self):  # ->double:
        return super().getPrice()

    def setPrice(self, price):
        super().setPrice(price)

    def getTypeOfAntenna(self) -> str:
        return self.typeOfAntenna

    def getOS(self) -> str:
        return self.OS

    def __eq__(self, other) -> bool:
        if not isinstance(other, Smartphone):
            raise TypeError("Операнд справа должен иметь тип Smartphone")
        return super().__eq__(other) and self.typeOfAntenna == other.getTypeOfAntenna() and self.OS == other.getOS()

    def __str__(self) -> str:
        return f"[{super().__str__()}, typeOfAntenna={self.typeOfAntenna}, OS={self.OS}]"

    def charge(self):
        print("И снова на зарядку")

File: HoWo5_6/Phone/Smartphone.java
package HoWo5_6.Phone;

public abstract class Smartphone extends Phone{
    private String typeOfAntenna;
    private String OS;



    public Smartphone(String model, double sizeDisplay, int akb, double price, String typeOfAntenna, String oS) {
        super(model, sizeDisplay, akb, price);
        this.typeOfAntenna = typeOfAntenna;
        this.OS = oS;
    }

    public void loginToTheInternet(){
        System.out.println("вы вошли в интернет, Добро пожаловать!!!)");
    }

    public void downloadApp(){
        System.out.println("очередной софт установлен");
    }



    public String getModel() {
        return super.getModel();
    }



    public double getSizeDisplay() {
        return super.getSizeDisplay();
    }



    public int getAkb() {
        return super.getAkb();
    }



    public double getPrice() {
        return super.getPrice();
    }


    public void setPrice(double price) {
        super.setPrice(price);
    }



    public String getTypeOfAntenna() {
        return typeOfAntenna;
    }



    public String getOS() {
        return OS;
    }

    @Override
    public String toString() {
        return "[" + super.toString() +
                ", typeOfAntenna=" + typeOfAntenna + ", OS=" + OS + "]";
    }

    @Override
    public void charge() {
        System.out.println("И снова на зарядку");
    }


}
